content/drafts/sphinx.md

```diff
@@ -0,0 +1,447 @@
+be given the message, the HMAC and the signer’s public key and can verify that 
+the owner of the private key corresponding to the given public key did in fact 
+create this HMAC.
+
+HMACs will be used throughout the sphinx packet construction so that each hop 
+can verify that the contents of the packet (the message) has not been tampered 
+with. Since the HMAC can only be derived by the sender, Alice, each HMAC for 
+each hop must be present in the onion from the beginning. This will become more 
+clear with the example. 
+
+# XOR
+
+The [XOR][xor] (or Exclusive-OR) operation is a bitwise operation where the 
+result is 1 if one _and only one_ of the two bits being operated on is 1. In 
+other words, the result of the XOR operation is only 1 if the two bits being 
+operated on differ. The XOR truth table can be seen below. It shows the result 
+(`C`) after performing the XOR operation on two bits, `A` and `B`, for each of 
+the various combinations.
+
+![](/onion/xor-1.png#center)
+
+The next diagram demonstrates an interesting property of XOR that we will be 
+making use of in the sphinx packet construction later in this post. It shows 
+that taking the result of the above operation, `C`, and XORing it with `A` 
+produces `B`. Similarly, the XOR of `C` and `B` produce `A`.
+
+![](/onion/xor-2.png#center)
+
+To really nail down the idea, let’s look at some examples: In example 1 below, 
+you can see that if you XOR a packet with itself, it produces a zero byte 
+array. So XORing something with itself essentially destroys information. 
+Example 2 shows that if you take a packet and XOR it with a zero byte array of 
+the same length, then it produces the original packet.
+
+![](/onion/xor-3.png#center)
+
+Ok let’s see some more useful and interesting examples:
+
+![](/onion/xor-4.png#center)
+
+Example 3 shows that if you take a packet and XOR it with a random byte stream, 
+then you get the encrypted form of the packet. Example 4 shows that if you then 
+take that encrypted packet and once again XOR it with the _same_ byte stream, 
+then you are once again left with the clear text. The sphinx packet 
+construction makes heavy use of XOR for encrypting a clear text packet using a 
+pseudo random byte stream.
+
+# Sender preparation
+
+The example used in this post continues where the last post ended off. 
+Basically we have the following route where Alice is the sender and Dave is the 
+recipient. Alice essentially just needs to deliver a payload to each hop 
+without leaking too much data about the route.
+
+![](/onion/1-path.png#center)
+
+Alice has a set of payloads that she wants to communicate to each hop. When the 
+payloads are put into the onion packet, they are prefixed with a length and 
+postfixed with a 32 HMAC. 
+
+![](/onion/5-payloads.png#center)
+
+Alice now goes ahead and generates a `session_key` and uses this to derive the 
+chain of [ephemeral keys](#ephemeral-keys). 
+
+![](/onion/3-ephemeral_keys.png#center)
```

Comment:
The line directly above the yellow box reads $e_{AD} = bf * e_{AD}$ but I think it should read $e_{AD} = bf * e_{AC}$ instead.

Also, it wasn't immediately clear to me that `bf` means blinding factor, as that's not used anywhere directly.

Reply:
fixed! thanks